src/serverlist.cpp

```diff
@@ -167,31 +167,52 @@ CServerListManager::CServerListManager ( const quint16  iNPortNum,
 
 void CServerListManager::SetCentralServerAddress ( const QString sNCentServAddr )
 {
+    // if the address has not actually changed, do nothing
+    if ( sNCentServAddr == strCentralServerAddress ) {
+        return;
+    }
+
+    // if we are registered to a custom directory server, unregister before updating the name
+    if ( eCentralServerAddressType == AT_CUSTOM && GetSvrRegStatus() == SRS_REGISTERED ) {
```

Comment:
Why only if it's custom?  Again, the comment says what.  Tell me why....

Reply:
It should be setting the actual address, rather than the enum value.  It should be what's doing the registration.  Picking one of the predefined addresses should call this method, really...